### ISSUE TITLE
DataDriftTrigger: Reference and current data windows for drift detection

### ISSUE THREAD

**Issue by jenny011** [eth-easl/modyn#418]
Currently for drift detection, DataDriftTrigger uses the training data of the previous trigger as the reference data and all untriggered data as the current data. While the finetuned model has theoretically seen all triggering data of all previous triggers, such windowing method only compares the new data with the data of the latest trigger. Such windowing could result in the drift detection missing drifts that affect model accuracy. 
We would like to support different windowing. For example, use the training data of all the previous triggers as the reference data.

**Comment by robinholzi** [comment]
Done
